charts/spire/charts/spiffe-oidc-discovery-provider/templates/pre-delete-hook.yaml

```
@@ -72,4 +72,6 @@ spec:
           - deployment
           - {{ include "spiffe-oidc-discovery-provider.fullname" . }}
           - --wait
+        resources:
+          {{- toYaml .Values.global.deleteHooks.resources | nindent 10 }}
```

How about something like the following here and elsewhere:
```suggestion
        {{ with (dig "deleteHooks" "resources" (dict) .Values.global) }}
        resources:
          {{- toYaml . | nindent 10 }}
        {{ end }}
```

It would work when globals are not defined in the case where we are talking about having the option to install the individual charts standalone.

---

Each evaulation of a Helm value will return nil (nothing) provided it is referenced by a valid name.  Additionally, (nil).name yeilds (nil).  For this reason I would suggest the following approach to replace the often-confusing "dig"

Also I think "with (nil)" evaluates to no visitation of the loop body, please verify this.

{{ with ((.Values.global).deleteHooks).resources }} can replace both the approaches, and reads much nicer.